src/app/services/distance.service.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DistanceService {
+  //public dict? : Map<string, { latitude: number; longitude: number }>;
+  public dict = {};
+  constructor(private http: HttpClient) {
+    this.makeDict();
+  }
+  public makeDict() {
```

Comment:
Is there a reason this is public? Does it ever need to be called outside of the constructor?